os/fs/smartfs/smartfs_smart.c

```diff
@@ -205,6 +203,7 @@ static int smartfs_open(FAR struct file *filep, const char *relpath, int oflags,
 	/* Locate the directory entry for this path */
 
 	sf = (struct smartfs_ofile_s *)kmm_malloc(sizeof * sf);
+	memset(sf, 0, sizeof(struct smartfs_ofile_s));
```

Comment:
it's not necessary, I think

Reply:
Actually I added this to resolve a bug.

sf->entry.datlen was not cleared and has old values.
So in smartfs_write, it would cause malfunctioning.